distribution/xfer/download.go

```diff
@@ -346,14 +361,56 @@ func (ldm *LayerDownloadManager) makeDownloadFunc(descriptor DownloadDescriptor,
 				return
 			}
 
+			desc, err := descriptor.Desc()
+			if err != nil {
+				d.err = err
+				return
+			}
+
+			var layerData io.Reader
```

Comment:
We should use containerd's stream processor with containerd-integration
https://github.com/moby/moby/labels/containerd-integration

Reply:
I add  it at daemon/containerd/image_pull.go, but I don't know how to verify it. I use docker cli pull some image but it call PullImage func at  docker/daemon/images/image_pull.go instead of  daemon/containerd/image_pull.go @AkihiroSuda

Reply:
add  "features": {"containerd-snapshotter":true}  cat /etc/docker/daemon.json it will  work